src/ast/ast_validations.c

```diff
@@ -762,8 +762,8 @@ static AST_Validation _Validate_MERGE_Clauses(const AST *ast, char **reason) {
 		uint clause_idx = merge_clause_indices[i];
 
 		// Collect all entities that are bound before this MERGE clause.
-		for(uint j = start_offset; i < clause_idx; i ++) {
-			const cypher_astnode_t *clause = cypher_ast_query_get_clause(ast->root, i);
+		for(uint j = start_offset; j < clause_idx; j ++) {
```

Comment:
Oh no!

Reply:
I know! I can't believe this didn't trip us up before.